mathesar/api/db/viewsets/data_files.py

```diff
@@ -22,6 +23,13 @@ class DataFileViewSet(viewsets.GenericViewSet, ListModelMixin, RetrieveModelMixi
     filter_backends = (filters.DjangoFilterBackend,)
     filterset_class = DataFileFilter
 
+    @extend_schema(
+        request=DataFileSerializer,
+        responses={
+            status.HTTP_200_OK: DataFileSerializer,
+            status.HTTP_405_METHOD_NOT_ALLOWED: 'Custom error response for method not allowed',
```

Comment:
Shouldn't this be a response body? Also it would be better to wrap it inside `OpenApiResponse`. Look at this [test case](https://github.com/tfranzel/drf-spectacular/blob/1f7cf5d8342f5ed3f619afcf92fb43e3b4ff26cb/tests/test_regressions.py#L1744)